applications/ink_contract_multi-sig.md

```diff
@@ -0,0 +1,215 @@
+
+By undertaking this approach, we aim to gain a comprehensive understanding of the current landscape and identify the areas where we can introduce innovative solutions. This will facilitate the development of a robust and effective smart contract that can fulfill the project's requirements.
+
+One of the major limitations we highlighted in the proposal is the inability to add or remove owners of the multisig wallet once it is created. To address this, our design will incorporate methods in the smart contract that allows this flexibility.
+
+Our ink! multisig smart contract will maintain an array of wallet owners and a threshold number that will be used to validate the transactions. When a proposal to add or remove an owner is created, owners of the wallet will have to confirm the change. If the number of confirmations crosses the defined threshold, the change will be committed.
+
+#### <a name="step2"></a>2.- Multi-sig Mininimum Viable Product (MVP)
+
+As a next step, we will combine the insights gained from our research with our experience in implementing multi-sig solutions in EVM environments. Drawing on our expertise in multi-sig architecture and lessons learned from building on top of Safe, we aim to develop an MVP that not only incorporates multi-sig logic in an ink! smart contract, but also provides an exceptional user experience through a well-designed user interface.
+
+In considering the requirements of large multi-sig contracts, we will also evaluate the need for a backend that reduces transaction costs and improves user interactions. Through this approach, we are confident that we can develop a robust, user-friendly multi-sig solution that meets the project's goals and provides value to its users.
+
+#### UI Wireframes
+>Note: These are just wireframes, the final UI design will be done by our UI/UX designer.
+
+- Select type of contract call
+
+![type](https://github.com/protofire/Ecosystem-Grants/blob/assets/assets/image0.png)
+
+---
+
+- Provide address and metadata
+
+If you choose to interact with a custom contract you have to provide the following:
+
+![metadata](https://github.com/protofire/Ecosystem-Grants/blob/assets/assets/image1.png)
+
+---
+
+- Choose the method and provide parameters
+
+![method](https://github.com/protofire/Ecosystem-Grants/blob/assets/assets/image2.png)
+
+---
+
+To summarize, what we will do:
+
+- Build an ink! smart contract-based multi-sig that interacts with the Contract Pallet. 
+- Develop a user-friendly front-end for interacting with the smart contract.
+- Evaluate the need for a backend to reduce transaction costs and improve user interactions in large multi-sig contracts, presenting a report with pros and cons of it.
+
+- Technology stack: Ink!, Rust, React, NextJS, TypeScript 
```

Comment:
For React, please take a look at [useink](https://github.com/paritytech/useink). There should be a major release in the next week or two. React Hooks for Wasm contracts